Include/internal/pycore_object.h

```diff
@@ -159,6 +159,24 @@ static inline void _Py_RefcntAdd(PyObject* op, Py_ssize_t n)
 }
 #define _Py_RefcntAdd(op, n) _Py_RefcntAdd(_PyObject_CAST(op), n)
 
+static inline int _Py_Reuse_Immutable_Object(PyObject *ob) {
```

Comment:
Swift calls this ["isKnownUniquelyReferenced'](https://developer.apple.com/documentation/swift/isknownuniquelyreferenced(_:)-98zpp), which seems like a reasonable name. I don't think it's important if `ob` is immutable or not in this context.

```suggestion
// Checks if an object has a single, unique reference. If the caller holds a
// unique reference, it may be able to safely modify the object in-place.
static inline int
_PyObject_IsUniquelyReferenced(PyObject *ob)
{
```